pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -304,6 +313,28 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req MetricsQueryReques
 		}
 	}
 
+	samplesProcessed := queryStats.LoadSamplesProcessedPerStep()
```

Comment:
nit: can we tuck these new changes away in a function? `Do` as broad as it sounds, is already doing a lot